rfcs/0145-doc-strings.md

```diff
@@ -211,17 +217,58 @@ int = /**Doc A*/1;
 # Dynamic attribute
 ```
 
+#### `Let .. in ..` binding
+
+```nix
+let
+ /** Documentation for the id function*/
+     ↓
+ a = x: x;
+in
+ a
+
+# Documentation can still be retrieved.
+```
+
 #### Partial lambda functions
 
 ```nix
 /**Generic doc for all partial lambda functions*/
 ↓1 ↓2          ↓3
-x: ({y, ...}:  z: x + y) * z;
+x: ({y, ...}:  z: x + y * z)
 
 # All partial functions can share the same placement with the outermost lambda.
-# 1 -> Lambda Value `x: ({y, ...}: z: x + y) * z` 
-# 2 -> Lambda Value `({y, ...}: z: x + y) * z`    
-# 3 -> Lambda Value `z: x + y) * z`               
+# 1 -> Lambda Value `x: ({y, ...}: z: x + y * z)` 
+# 2 -> Lambda Value `({y, ...}: z: x + y * z)`    
+# 3 -> Lambda Value `z: x + y * z)`               
+```
+
+> Allows for fallback documentation in case documentation on the partially applied function cannot be written or retrieved.
+
+#### Lambda formals
+
+```nix
+/**Doc for the whole lambda function*/
+{
+ /**Doc for attribute 'a'*/
+ a
+}:
+ a              
+```
+
+```nix
+/**Doc for the whole lambda function*/
+{
+ /**Doc for attribute 'a'*/
+ a,
+ ...
+}@args:
+ a + args.b;
+
+# `...` and `@args` cannot (yet) be documented.
```

Comment:
why cannot this be documented?

Reply:
How could this be documented? Do you think implementation and **retrieval** works? If yes then we'll change it. Currently i dont know how this could work.

Reply:
Don't know how implementation retrieval works, I just think it is unnecessary to explicitly *forbid* this. The sentence is describing implementation detail (we cannot retrieve this comment).

Maybe we can write something like kwargs documentation here (e.g. python docs).

Reply:
Maybe remove this particular example? We have yet to have any POCs proving or showing the implications.